### ISSUE TITLE
High CPU mysql when running files_external:notify

### ISSUE THREAD

**Issue by Rick26L** [nextcloud/server#9105]
### Steps to reproduce
1. Mount an external SMB Storage 
2. Start occ command files_external:notify for the external storage 
3. 

### Expected behaviour
The notify command should track changes which are made on the external storage and write the outdated files and folders it in the database. 
Then the occ files:scan --unscanned command should trigger to scan the files and folder which are marked as outdated in the database, that the files and folders where 
updated in the nextcloud windows sync client

### Actual behaviour
The notify command starts detecting changes made on the external storage (able to see when using -vvv option). 
The CPU Load goes through 80-100% for the mysql process. 
The files:scan --unscanned outputs: 
Starting scan for user 1 out of 1 (*username*)
+---------+-------+--------------+
| Folders | Files | Elapsed time |
+---------+-------+--------------+
| 0       | 0     | 00:00:00     |
+---------+-------+--------------+

### Server configuration

**Operating system**:
Ubuntu 16.0.4

**Web server:**
Apache Webserver 2.4.18

**Database:**
MySQL 10.2.13

**PHP version:**
PHP 7.0

**Nextcloud version:** 
12.0.6

**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated from NC 12.0.3

**Where did you install Nextcloud from:**
Tech and Me script

**Signing status:**
No errors have been found. 

**List of activated apps:**
Enabled:
  - activity: 2.5.2
  - admin_audit: 1.2.0
  - admin_notifications: 1.0.1
  - apporder: 0.4.1
  - bruteforcesettings: 1.0.3
  - comments: 1.2.0
  - dav: 1.3.1
  - drawio: 0.8.9
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_accesscontrol: 1.2.5
  - files_clipboard: 0.7.1
  - files_downloadactivity: 1.1.1
  - files_external: 1.3.0
  - files_inotify: 0.1.1
  - files_pdfviewer: 1.1.2
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - firstrunwizard: 2.1
  - impersonate: 1.0.3
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextant: 1.0.9
  - nextcloud_announcements: 1.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - onlyoffice: 1.3.0
  - password_policy: 1.2.2
  - previewgenerator: 1.0.9
  - provisioning_api: 1.2.0
  - quota_warning: 1.1.1
  - ransomware_protection: 1.0.5
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - socialsharing_email: 1.0.3
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - updatenotification: 1.2.0
  - user_external: 0.4
  - user_ldap: 1.2.1
  - workflowengine: 1.2.0
Disabled:
  - calendar
  - contacts
  - encryption
  - gallery
  - ojsxc
  - passman
  - richdocuments
  - spreedme
  - workin2gether

**Nextcloud configuration:**
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "*local-IP*",
            "cloud.*domain*.de",
            "office.*domain*.de"
        ],
        "datadirectory": "\/var\/ncdata",
        "overwrite.cli.url": "https:\/\/cloud.*domain*.de\/",
        "dbtype": "mysql",
        "version": "12.0.6.1",
        "dbname": "nextcloud_db",
        "dbhost": "localhost",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "oc989awy4zxd",
        "maintenance": false,
        "mail_smtpmode": "smtp",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": true,
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "\/var\/run\/redis\/redis.sock",
            "port": 0,
            "timeout": 0,
            "dbindex": 0,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "htaccess.RewriteBase": "\/",
        "loglevel": 2,
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "*username*",
        "mail_domain": "*domain*.de",
        "mail_smtphost": "mail.*domain*.de",
        "mail_smtpport": "25",
        "theme": "",
        "updater.release.channel": "stable"
    }
}

**Are you using external storage, if yes which one:** local/smb/sftp/...
smb

**Are you using encryption:** 
no 

**Are you using an external user-backend, if yes which one:** 
LDAP AD integration

### Client configuration
**Browser: Chrome 65 / Firefox 59

**Operating system:**
Windows 10 x64

**Comment by Hadessohn** [comment]
Could be the same issue as #9042

**Comment by dohlin** [comment]
@Rick26L I can't guarantee you're seeing the exact same issue, but try disabling W2G2 and rebooting.  Feel free to see the steps I took to arrive at the conclusion that at least in my case this app is what was causing my similar issue over in issue #9042.  Figured I'd post over here too just in case it helps!  Thanks!

**Comment by Rick26L** [comment]
Hi @dohlin Thank you for your feedback. I removed W2G2 rebootet and testet again. 
When I start the files_external:notify the CPU goes high to 80% but after a few minutes it goes down. Sometimes it goes to 10% but I think thats okay. 
So after removing W2G2 the files_external:notify command runs without pushing the CPU to the limit. 

But the Problem that still exists is when I run the occ files:scan --unscanned that no files are scanned. But the notify already listet some files which have changed. Maybe I'm doing something wrong. 

That's what I'm doing: 
1. Start the Notify 
-> sudo -u www-data php /var/www/nextcloud/occ files_external:notify --user *DomainUser* --password *password* -vvv -- 4
---> After I entered the Command I get the feedback "Self-test successful" and files which are changed are listet in the command window. 
2. Start the scan which should only scan files which are marked by the notify command
-> sudo -u www-data php /var/www/nextcloud/occ files:scan --unscanned --all
--> It starts the scanning process with the feedback that 0 files and 0 folder are scanned. 

Some Idea?

**Comment by dohlin** [comment]
@Rick26L Unfortunately I'm not sure on that - I honestly never have run either of those commands so I'm guessing that fundamentally our issues are different.  When I ran 'top' via terminal I could see that specifically the 'mysqld' service caused my processor to nearly instantly go from ~40-50% total usage give or take to 100% pegged on both cores.  Removing W2G2 fixed that issue.  Otherwise I simply have a NAS share locally CIFS mounted (via my /etc/fstab file) and I pointed Nextcloud's external storage support to that "local" mount point.  Sorry!
